src/expr/expr_manager_template.cpp

```diff
@@ -907,9 +886,6 @@ Type ExprManager::getType(Expr expr, bool check)
 }
 
 Expr ExprManager::mkVar(const std::string& name, Type type, uint32_t flags) {
-  Assert(NodeManager::currentNM() == NULL)
```

Comment:
How is this related to this PR?

Reply:
This is a rather bizarre assertion that checks whether there is a node manager in scope to determine if we are calling this method internally or externally.

After this PR, this distinction no longer holds, since NodeManager is always in scope whenever an SmtEngine has been created.